sale_global_discount/models/sale_order.py

```diff
@@ -94,44 +94,52 @@ def _compute_amounts(self):
         res = super()._compute_amounts()
         for order in self:
             order._check_global_discounts_sanity()
-            amount_untaxed_before_global_discounts = order.amount_untaxed
-            amount_total_before_global_discounts = order.amount_total
+            vals = {
+                "amount_untaxed_before_global_discounts": order.amount_untaxed,
+                "amount_total_before_global_discounts": order.amount_total,
+            }
             discounts = order.global_discount_ids.mapped("discount")
-            amount_discounted_untaxed = amount_discounted_tax = 0
-            for line in order.order_line:
-                discounted_subtotal = line.price_subtotal
-                if not line.product_id.bypass_global_discount:
-                    discounted_subtotal = self.get_discounted_global(
-                        line.price_subtotal, discounts.copy()
+            if discounts:
+                amount_discounted_untaxed = amount_discounted_tax = 0
+                order_lines = order.order_line.filtered(lambda x: not x.display_type)
+                for line in order_lines:
+                    discounted_subtotal = line.price_subtotal
+                    if not line.product_id.bypass_global_discount:
+                        discounted_subtotal = self.get_discounted_global(
+                            line.price_subtotal, discounts.copy()
+                        )
+                    amount_discounted_untaxed += discounted_subtotal
+                    discounted_tax = line.tax_id.compute_all(
+                        discounted_subtotal,
+                        line.order_id.currency_id,
+                        1.0,
```

Comment:
Why using 1.0 and not mimic Odoo method ? https://github.com/odoo/odoo/blob/45154cfdb044aea85ad72bb36348af7b77afd844/addons/sale/models/sale_order_line.py#L879